core/src/program/mod.rs

```diff
@@ -147,3 +151,11 @@ impl Kernel {
         &self.0
     }
 }
+
+// this is required by AIR as public inputs will be serialized with the proof
+impl Serializable for Kernel {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.0.len() as u64);
+        Digest::write_batch_into(&self.0, target)
```

Comment:
The length is unlikely to be greater than 256 here, so we could probably get away with using a single byte for the number of procedures. But for now, let's just put a TODO to come back to it later.

Reply:
Done; changed to `u16` for now to reflect the serialization limit of `assembly::serde`